core/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -269,9 +272,10 @@ public void writeTo(StreamOutput out) throws IOException {
     public static class Builder {
 
         private static final Set<String> VALID_FIELDS = Sets.newHashSet("template", "order", "mappings", "settings", "index_patterns");
-        static {
-            VALID_FIELDS.addAll(IndexMetaData.customPrototypes.keySet());
-        }
+//        static {
+        // TODO: add proto fields
```

Comment:
Oops. These are leftovers from when I was trying to convert custom index metadata to the new mode. I will restore this back.